nixos/modules/services/misc/preload.nix

```diff
@@ -0,0 +1,32 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.preload;
+  preload = pkgs.${pkgs.system}.preload;
+in {
+  meta = { maintainers = preload.meta.maintainers; };
+
+  options = {
+    services.preload = { enable = mkEnableOption (mdDoc "preload"); };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.preload = {
+      description = "preload daemon";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        EnvironmentFile = "${preload}/etc/conf.d/preload";
+        ExecStart = "${getExe preload} --foreground $PRELOAD_OPTS";
+        Type = "simple";
+        IOSchedulingClass = 3;
```

Comment:
Ok I will make a comment. The reason is that `IOSchedulingClass = 3;` causes preload to only execute if the cpu has nothing to do. This is needed because well the idea of preload is loading data into memory during the free time of the cpu to speed up things a bit.